README.md

```diff
@@ -0,0 +1,142 @@
+# Pactum
+## The HTTP-API specification sketchbook for pythonistas
+
+[![Circle CI](https://circleci.com/gh/olist/pactum.svg?style=svg)](https://circleci.com/gh/olist/pactum)
+
+```shell
+pip install pactum
+```
+
+With `Pactum` you can **specify** HTTP-APIs using pure python.
+
+Pactum is easy to use, easy to extend and easy to contribute:
+
+### Easy to use
+
+The only requirements to start writing an API specification with `pactum`
+is `pactum` package itself and some knowledge of python.
+
+```python
+import pactum
+
+class MyAPI(API):
+    name = 'My API'
+    versions = [...]
+```
+
+### Easy to extend
+
+Using the [visitor pattern](http://wiki.c2.com/?VisitorPattern) you can create
+exporters and extensions for any format or service you want.
+
+Take a look at [pactum/exporters/openapi.py](https://github.com/olist/pactum/blob/master/pactum/exporters/openapi.py).
+
+### Architecture
+
+Always keep [this diagram](https://github.com/olist/pactum/wiki/Architecture-Diagram) in mind when defining your APIs.
+
+
+### Tutorial
+
+Create a file called specs.py and start defining your API.
+
+You can define a `Resource` object for your API.
+
+```python
+from pactum import Action, API, Resource, Response, Version
+from pactum import fields, verbs
+
+class OrderResource(Resource):
+    fields = [
+        fields.IntegerField(name='code', required=True),
+        fields.TimestampField(name='created_at'),
+        fields.StringField(name='item')
+    ]
+resource = OrderResource()
+
+error_resource = Resource(
+    name = 'ErrorResource'
+    fields = [fields.StringField(name='error', required=False)]
+)
+```
+You can define any element of your specification by calling it directly as in
+`error_resource` or by class definition as in `MyResource` and then calling it.
+
+
+List resources are definitions of lists of the same resource.
+```python
+list_order_resource = ListResource(resource=resource)
```

Comment:
is it possible to pass a class instead of an instance here? IMHO it would be nice to do so because we won't have to instantiate each resource to link it with list resources or requests. for APIs with a lot of resources this might get tiresome and  maybe even kinda ugly.

Reply:
No. Instances > Classes in this case because we can provide functionalities or properties that dynamically adjust an specification (could be useful for, eg, export a swagger with an specific subset of spec to clients with restrict permissions).

I know that we could do that with classes but we have decided to use only instances to make (or force) only one way to create specification.

You don't need to instantiate each resource to link. You could create only one instance and use it everywhere (eg. like a Singleton)

Reply:
Even so if the user is obligated create instances and the API have a lot of resources I have the impression a bunch of boilerplate code is going to being implemented. And I'm not fond of boilerplate code :stuck_out_tongue: 

And to be clear my suggestion was to accept both instances and classes: instances for more complex cases which require any number of customization and classes for when you don't need that.

Reply:
I don't see the impact in boilerplate generation in class vs. instances scenario. Can you show some example? It could be something that we missed at pactum design or one thing that need more/detailed explanation to make clear why we choose to use instances over classes (and why we believe that allow both is not a "good idea")...